torch/csrc/jit/fusion_compiler.cpp

```diff
@@ -206,10 +212,11 @@ std::string encodeRHS(Node * n) {
     {aten::div, "${0} / ${1}"},
     {aten::eq, "${0} == ${1}"},
     {aten::fmod, "fmodf(${0}, ${1})"},
-    {aten::ge, "${0} >= ${1})"},
+    {aten::ge, "(${0} >= ${1})"},
     {aten::gt, "${0} > ${1}"},
-    {aten::le, "${0} <= ${1})"},
+    {aten::le, "(${0} <= ${1})"},
     {aten::lt, "${0} < ${1}"},
+    {aten::type_as, "(${0})"}, //everything is implicitly convertible to float
```

Comment:
This also feels quite wrong. We really shouldn't embed e.g. operations on integral tensors into float pointwise kernels, because not all values are representable. That's why the fuser pass is so restrictive

Reply:
type_as is not inserted by fuser, it comes from user graph. If user calls type_as on a tensor containing unrepresentable ints, same thing would happen, fusion or not.

Reply:
Halfs are implicitly convertible to float? :)

Reply:
Halfs are a whole different can of worms that will be handled by a different PR :-) For now, anything with halfs won't even compile with nvrtc, because nvrtc is that awesome.